crates/ruff/src/rules/ruff/rules/collection_literal_concatenation.rs

```diff
@@ -54,80 +54,135 @@ enum Kind {
-    let (kind, splat_element, other_elements, splat_at_left, ctx) =
-        match (left.as_ref(), right.as_ref()) {
-            (
-                Expr::List(ast::ExprList {
-                    elts: l_elts,
-                    ctx,
-                    range: _,
-                }),
-                _,
-            ) => (Kind::List, right, l_elts, false, ctx),
-            (
-                Expr::Tuple(ast::ExprTuple {
-                    elts: l_elts,
-                    ctx,
-                    range: _,
-                }),
-                _,
-            ) => (Kind::Tuple, right, l_elts, false, ctx),
-            (
-                _,
-                Expr::List(ast::ExprList {
-                    elts: r_elts,
-                    ctx,
-                    range: _,
-                }),
-            ) => (Kind::List, left, r_elts, true, ctx),
-            (
-                _,
-                Expr::Tuple(ast::ExprTuple {
-                    elts: r_elts,
-                    ctx,
-                    range: _,
-                }),
-            ) => (Kind::Tuple, left, r_elts, true, ctx),
-            _ => return,
-        };
+    let (kind, splat_element, other_elements, splat_at_left, ctx) = match (&new_left, &new_right) {
+        (
+            Expr::List(ast::ExprList {
+                elts: l_elts,
+                ctx,
+                range: _,
+            }),
+            _,
+        ) => (Kind::List, &new_right, l_elts, false, ctx),
+        (
+            Expr::Tuple(ast::ExprTuple {
+                elts: l_elts,
+                ctx,
+                range: _,
+            }),
+            _,
+        ) => (Kind::Tuple, &new_right, l_elts, false, ctx),
+        (
+            _,
+            Expr::List(ast::ExprList {
+                elts: r_elts,
+                ctx,
+                range: _,
+            }),
+        ) => (Kind::List, &new_left, r_elts, true, ctx),
+        (
+            _,
+            Expr::Tuple(ast::ExprTuple {
+                elts: r_elts,
+                ctx,
+                range: _,
+            }),
+        ) => (Kind::Tuple, &new_left, r_elts, true, ctx),
+        _ => return None,
+    };
 
     // We'll be a bit conservative here; only calls, names and attribute accesses
     // will be considered as splat elements.
-    if !(splat_element.is_call_expr()
+    let mut new_elts;
+    if splat_element.is_call_expr()
         || splat_element.is_name_expr()
-        || splat_element.is_attribute_expr())
+        || splat_element.is_attribute_expr()
     {
-        return;
+        new_elts = make_splat_elts(splat_element, other_elements, splat_at_left);
+    }
+    // If the splat element is itself a list/tuple, insert them in the other list/tuple.
+    else if splat_element.is_list_expr() || splat_element.is_tuple_expr() {
+        new_elts = other_elements.clone();
+
+        let mut elts = match splat_element {
```

Comment:
How would this deal with `(1,) + [2]` or `[1] + (2,)`?

Reply:
I did think about this, but 

Running ruff on a file with:
```python
a = (1,) + [2]
b = (1, 2, 3) + (4,)
c = (7, 8, 9) + b
```

The current version outputs a warning only for the last line, whereas the PR in its current state outputs a warning for all three lines. While a warning on the first line is clearly undesired (that's a typing error), should the one on the second line be kept ?